Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/devicetime/DeviceTimeHandler.kt

```diff
@@ -0,0 +1,54 @@
+package de.rki.coronawarnapp.appconfig.devicetime
+
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.appconfig.devicetime.ui.IncorrectDeviceTimeNotification
+import de.rki.coronawarnapp.main.CWASettings
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.sample
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DeviceTimeHandler @Inject constructor(
+    @AppScope private val scope: CoroutineScope,
+    private val appConfigProvider: AppConfigProvider,
+    private val notification: IncorrectDeviceTimeNotification,
+    private val cwaSettings: CWASettings
+) {
+
+    fun launch() {
+        appConfigProvider.currentConfig
+            .onStart { Timber.tag(TAG).d("Observing device time.") }
+            // If we don't delay emissions, we will consume the event before the UI "wasDeviceTimeIncorrectAcknowledged"
+            // ForegroundState.isInForeground has a day when opening the app.
+            .sample(5000)
+            .onEach {
+                Timber.tag(TAG).v("Current device time offset is: %dms", it.localOffset.millis)
+                if (it.isDeviceTimeCorrect) {
+                    Timber.v("Dismissing any notification, device time is correct again.")
+                    notification.dimiss()
```

Comment:
I think we also need to reset the "notification was shown" flag

"Once the device time has been corrected, the persisted information about the already shown notification is reset, so that the next time the device time is wrong again, the user gets again a notification"

If I time travel, I receive the notification once. After reset to the current time, I let the app be in that state for 10 minutes, then time-traveled again but did not receive a notification/pop-up.